collab-document/src/document.rs

```diff
@@ -523,6 +513,35 @@ impl DocumentBody {
     })
   }
 
+  fn write_from_document_data(
```

Comment:
The only concern here is that it could quickly increase the document size. For example, a single character change might overwrite the entire document content.